.circleci/config.yml

```
@@ -376,6 +376,40 @@ jobs:
             npm install
             node scripts/index_search_content.js
 
+  # generate server ts definitions
+  gen-ts: 
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      <<: *ENVIRONMENT
+    steps:
+      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - "c6:96:98:82:dc:04:6c:39:dd:ac:83:05:e3:15:1c:98"
+      - attach_workspace:
+          at: .
+      - run:
+          name: gen/ts
+          command: |
+            # check if there are any changes in internal/server/server.go
+            # if there are, we commit the ui static asset file
+            # HEAD^! is shorthand for HEAD^..HEAD (parent of HEAD and HEAD)
+            if ! git diff --quiet --exit-code HEAD^! internal/server/proto/server.proto; then
+              git config --local user.email "hashicorp-ci@users.noreply.github.com"
+              git config --local user.name "hashicorp-ci"
+              short_sha=$(git rev-parse --short HEAD)
+              go generate ./internal/server
+              make gen/ts
+              git add ui/lib
+              git commit -m "auto-updated ui/lib/* from commit ${short_sha}"
+              git push origin circle-ts-gen
```

```suggestion
              git push origin circle-ts-gen-${short_sha}
```

In the case where lingering PRs aren't addressed this might cause some headache if they all push to the same branch.

---

I think you need to checkout to this branch too before comitting/pushing?

---

yeah, I'll need to checkout a new branch and do that. I wasn't there yet. Mostly focused on getting the gen/ts step working first

---

Yep! Sorry, I jumped the gun a bit on my review! ❤️